### PR TITLE
WIP - #2568

### PR DESCRIPTION
**Pull request by ywarnier**
Work in progress with the private data treatment information

### CODE REVIEW

File: main/admin/legal_list.php [hunk 1]
```diff
@@ -8,6 +8,8 @@
  */
 $cidReset = true;
 require_once __DIR__.'/../inc/global.inc.php';
+use Chamilo\CoreBundle\Entity\Repository\LegalRepository;
```
Comment (jmontoyaa) [inline main/admin/legal_list.php]: The "use" should be in the top before global.inc check other files.

File: main/social/personal_data.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+api_block_anonymous_users();
+if (!api_get_configuration_value('enable_gdpr')) {
+    api_not_allowed(true);
+}
+
+$userId = api_get_user_id();
+
+$entityManager = Database::getManager();
+/** @var UserRepository $repository */
+$repository = $entityManager->getRepository('ChamiloUserBundle:User');
```
Comment (jmontoyaa) [inline main/social/personal_data.php]: You can use "api_get_user_entity($userId)", There are also "api_get_course_entity()" and "api_get_session_entity() "functions.

File: main/social/personal_data.php [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+$icon = Display::return_icon('export_excel.png', get_lang('Export'), null,ICON_SIZE_MEDIUM);
+$personalData['data_export_icon'] = $icon;
+$personalData['permissions'] = $termsAndConditionsAcceptance;
+$personalData['responsible'] = api_get_setting('personal_data_responsible_org');
+
+$em = Database::getManager();
+/** @var LegalRepository $legalTerms */
+$legalTermsRepo = $em->getRepository('ChamiloCoreBundle:Legal');
+//Get data about the treatment of data
+$treatmentTypes = [
```
Comment (jmontoyaa) [inline main/social/personal_data.php]: If the types (101, 102, etc) are hardcode values they should be define in some library legal.lib, or better in the entity ...

File: src/Chamilo/CoreBundle/Entity/TrackELogin.php [hunk 4]
```diff
@@ -18,9 +18,14 @@ class TrackELogin
     /**
      * @var int
```
Comment (jmontoyaa) [inline src/Chamilo/CoreBundle/Entity/TrackELogin.php]: Just as a reminder, this "@ orm" changes mean changes in the database. So they are not allowed in 1.11.x.
You can remove the "@" and then doctrine will ignore those changes, then in configuration.dist you can add a note to add the "@".

**Comment by ywarnier** [comment]
@jmontoyaa I pushed this to a branch of this project directly. Could you clean up for the two comments you mentioned and for which I have some doubts still? (won't have time to do that properly and I want this to move forward quickly and there's still some work). Work on the branch with me for now, then we'll merge in a few days.

**Comment by ywarnier** [comment]
Moved to #2606

**Pull request closed without merging**